genny/auth/auth.go

```diff
@@ -0,0 +1,125 @@
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gobuffalo/buffalo/meta"
+	"github.com/gobuffalo/genny"
+	"github.com/gobuffalo/genny/movinglater/attrs"
+	"github.com/gobuffalo/genny/movinglater/gotools"
+	"github.com/gobuffalo/genny/movinglater/plushgen"
+	"github.com/gobuffalo/packr"
+	"github.com/gobuffalo/plush"
+	"github.com/pkg/errors"
+)
+
+func extraAttrs(args []string) []string {
+	var names = map[string]string{
+		"email":    "email",
+		"password": "password",
+		"id":       "id",
+	}
+
+	var result = []string{}
+	for _, field := range args {
+		at, _ := attrs.Parse(field)
+		field = at.Name.Underscore().String()
+
+		if names[field] != "" {
+			continue
+		}
+
+		names[field] = field
+		result = append(result, field)
+	}
+
+	return result
+}
+
+var fields attrs.Attrs
+
+// New actions/auth.go file configured to the specified providers.
+func New(args []string) (*genny.Generator, error) {
+	g := genny.New()
+
+	var err error
+	fields, err = attrs.ParseArgs(extraAttrs(args)...)
+	if err != nil {
+		return g, errors.WithStack(err)
+	}
+
+	if err := g.Box(packr.NewBox(filepath.Join("..", "auth", "templates"))); err != nil {
+		return g, errors.WithStack(err)
+	}
+
+	ctx := plush.NewContext()
+	ctx.Set("app", meta.New("."))
+	ctx.Set("attrs", fields)
+
+	g.Transformer(plushgen.Transformer(ctx))
+	g.Transformer(genny.NewTransformer(".html", newUserHTMLTransformer))
+	g.Transformer(genny.NewTransformer(".fizz", migrationsTransformer))
+
+	g.RunFn(func(r *genny.Runner) error {
+		path := filepath.Join("actions", "app.go")
+		file, err := os.Open(path)
```

Comment:
use `r.FindFile` instead. it will look for the file in a cache if previously accessed, or will look to disk to find it.

Reply:
All set :)